yarn-project/aztec-nr/aztec/src/note/lifecycle.nr

```diff
@@ -30,6 +35,24 @@ pub fn create_note<Note, N>(
     assert(notify_created_note(storage_slot, preimage, inner_note_hash) == 0);
 
     context.push_new_note_hash(inner_note_hash);
+
+    let serialize = note_interface.serialize;
+
+    if broadcast {
+        let get_owner = note_interface.get_owner;
+        let maybe_owner = get_owner(*note);
+        assert(maybe_owner.is_some(), "Note owner is must be set when broadcast flow is triggered.");
+
+        let owner = maybe_owner.unwrap_unchecked();
+        let owner_key = get_public_key(owner);
```

Comment:
Related to my other comment, the note owner is also something that gets a little strange to me, since you can in many cases implicitly handle it by the way you structure your storage instead of actually needing the value as part of the note.

If we start adding things that might not be needing and this is also how the PXE expects things etc, seems like you could end up with a split between nodes and what notes they can read 😅

Reply:
yes, it's weird. I also just realized that the [ProfileNote](https://github.com/AztecProtocol/aztec-packages/blob/master/yarn-project/noir-contracts/src/contracts/docs_example_contract/src/types/profile_note.nr) does not have it and it feels incorrect to introduce it there.

I think it would be better to rework this and as you proposed add the emit_encrypted_log to the note interface. But to do that I would still need to somehow get hold of the encryption public key but the dev would have the freedom to decide how that will be done so it seems like an improvement.